internal/hardware/validator.go

```diff
@@ -150,6 +151,28 @@ func (v *validator) DiskIsEligible(ctx context.Context, disk *models.Disk, infra
 		}
 	}
 
+	// We are not allow iSCSI disk with multipath holder
+	if disk.DriveType == models.DriveTypeISCSI {
+		var multipathDiskName string
+		for _, inventoryDisk := range allDisks {
+			if inventoryDisk.DriveType == models.DriveTypeMultipath {
+				multipathDiskName = inventoryDisk.Name
+				break // No need to continue searching once the multipath disk is found
+			}
+		}
+
+		if multipathDiskName != "" {
+			for _, inventoryDisk := range allDisks {
+				if inventoryDisk.Name == disk.Name {
+					if funk.ContainsString(strings.Split(inventoryDisk.Holders, ","), multipathDiskName) {
```

Comment:
what contains `holders` exactly? If it's always a multipath drive, can't we just do something like:
```
	if disk.DriveType == models.DriveTypeISCSI && disk.Holders != "" {
               notEligibleReasons = append(notEligibleReasons, iSCSIWithMultipathHolder)
        }
```

Reply:
@adriengentil The reason I didn't implement that approach is because I wasn't sure if an iSCSI disk can have holders other than multipath. If you can confirm that it cannot, I can use your suggestion, as it would simplify everything.

Reply:
We can look into the code to understand what fills the `Holders` field

Reply:
@adriengentil https://github.com/openshift/assisted-installer-agent/blob/master/src/inventory/disks.go#L262-L277
It doesn't imply what drive type the holders are.

Reply:
safer to check then :+1: